interfaces/ros1/wavemap_ros/src/inputs/depth_image_input.cc

```diff
@@ -0,0 +1,182 @@
+#include "wavemap_ros/inputs/depth_image_input.h"
+
+#include <cv_bridge/cv_bridge.h>
+#include <opencv2/core/eigen.hpp>
+#include <sensor_msgs/PointCloud.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <sensor_msgs/point_cloud_conversion.h>
+#include <wavemap/core/integrator/projective/projective_integrator.h>
+#include <wavemap/core/utils/iterate/grid_iterator.h>
+#include <wavemap/core/utils/print/eigen.h>
+#include <wavemap/core/utils/profiler_interface.h>
+
+namespace wavemap {
+DECLARE_CONFIG_MEMBERS(DepthImageInputConfig,
+                      (topic_name)
+                      (topic_queue_length)
+                      (measurement_integrator_names)
+                      (processing_retry_period)
+                      (max_wait_for_pose)
+                      (sensor_frame_id)
+                      (image_transport_hints)
+                      (depth_scale_factor)
+                      (time_offset)
+                      (projected_pointcloud_topic_name));
+
+bool DepthImageInputConfig::isValid(bool verbose) const {
+  bool all_valid = true;
+
+  all_valid &= IS_PARAM_NE(topic_name, std::string(""), verbose);
+  all_valid &= IS_PARAM_GT(topic_queue_length, 0, verbose);
+  all_valid &= IS_PARAM_FALSE(measurement_integrator_names.empty(), verbose);
+  all_valid &= IS_PARAM_GT(processing_retry_period, 0.f, verbose);
+  all_valid &= IS_PARAM_GE(max_wait_for_pose, 0.f, verbose);
+
+  return all_valid;
+}
+
+DepthImageInput::DepthImageInput(const DepthImageInputConfig& config,
+                                 std::shared_ptr<Pipeline> pipeline,
+                                 std::shared_ptr<TfTransformer> transformer,
+                                 std::string world_frame, ros::NodeHandle nh,
+                                 ros::NodeHandle nh_private)
+    : InputBase(config, std::move(pipeline), std::move(transformer),
+                std::move(world_frame), nh, nh_private),
```

Comment:
Is there a particular reason to std::move the world_frame given it's just a std::string and expected as such by InputBase?